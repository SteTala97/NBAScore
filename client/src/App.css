.navigation {
  position: fixed;
  display: flex;
  top: 0;
  z-index: 10;
  align-items: center;
  height: 60px;
  width: 100%;
  padding: 0.5rem 0rem;
  background-color: #fff;
  color: black;
  box-shadow: 0 2px 2px 2px rgba(9, 9, 9, 0.23);
}

.div-nav{
  margin-top: 0;
  margin-bottom: 5em;
}

.brand-name {
  text-decoration: none;
  color: black;
  font-size: 1.3rem;
  margin-left: 1rem;
}

.navigation-menu {
  margin-left: auto;
  z-index: 2;
}

.navigation-menu ul {
  display: flex;
  padding: 0;
}

.navigation-menu li {
  list-style-type: none;
  margin: 0 1rem;
}

.navigation-menu li a {
  text-decoration: none;
  color:black;;
  display: block;
  width: 100%;
}

.hamburger {
  border: 0;
  height: 40px;
  width: 40px;
  padding: 0.5rem;
  border-radius: 50%;
  background-color: black;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  position: absolute;
  top: 50%;
  right: 25px;
  transform: translateY(-50%);
  display: none;
}

.hamburger:hover {
  background-color: grey;
}

  .hamburger {
    display: block;
  }

  .navigation-menu ul {
    flex-direction: column;
    position: absolute;
    top: 60px;
    left: 0;
    width: 100%;
    height: calc(100vh - 60px);
    background-color: white;
    border-top: 1px solid black;
    display: none;
    overflow: scroll; /*<- aggiunto per avere scroll-bar verticale nel menu di navigazione*/
  }

  .navigation-menu li {
    text-align: center;
    margin: 0;
  }

  .navigation-menu li a {
    color: black;
    width: 100%;
    padding: 1.5rem 0;
  }

  .navigation-menu li:hover {
    background-color: #eee;
  }

  .navigation-menu.expanded ul {
    display: block;
  }


.selectable-team {
  cursor: pointer;
}
