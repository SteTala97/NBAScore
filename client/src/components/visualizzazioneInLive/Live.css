.LiveTable{
  margin: 0 auto;
  margin-top: 35px;
  margin-bottom: 50px;
}