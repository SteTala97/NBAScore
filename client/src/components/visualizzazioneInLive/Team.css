.team-text {
  max-width: 25vh; /* 25% del totale*/ 
  justify-content: center;
}
