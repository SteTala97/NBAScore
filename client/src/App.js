import React from 'react';
import { useState } from "react";
import {
  HashRouter as Router,
  Routes,
  Route,
  Link,
  Navigate
} from "react-router-dom";


import './App.css';

import GamesPerDateView from './components/ricercaPartitePerData/GamesPerDateView.js';
import Live from './components/visualizzazioneInLive/Live.js';
import StandingView from './components/visualizzazioneClassifiche/standing-view.js';
import GamesPerTeam from './components/partitePerSquadra/GamesPerTeam.js';
import SearchPlayer from './components/ricercaGiocatorePerNome/SearchPlayer.js';
import SearchTeam from './components/ricercaSquadraPerNome/SearchTeam.js';
import StandingsPerTeamView from './components/visualizzazioneClassifichePerSquadra/StandingsPerTeamView.js';

import { createTheme, ThemeProvider } from "@mui/material/styles";
import WinningTeamView from './components/formazioneVincente/WinningTeamView.js';

const theme = createTheme({
  typography: {
    fontFamily: [
      "system-ui",
      "-apple-system",
      "system-ui",
      "Helvetica Neue",
      "Helvetica",
      "Arial",
      "sans-serif"
    ].join(",")
  }
});

function App() {
  const [isNavExpanded, setIsNavExpanded] = useState(false)
  // teamClicked e teamName servono per gestire la visualizzazione di TeamInfo nelle varie altre componenti 
  const [teamClicked, setTeamClicked] = useState(false);
  const [teamName, setTeamName] = useState("");


  return (

    
    <ThemeProvider theme={theme}>
      <div>
      <Router>
      <div className='div-nav'>
      <nav className="navigation">
      <a href="/" className="brand-name">
        NBAScore
      </a>
      <button
        className="hamburger"
        onClick={() => {
          setIsNavExpanded(!isNavExpanded);
        }}
      >
        {/* icon from Heroicons.com */}
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-5 w-5"
          viewBox="0 0 20 20"
          fill="white"
        >
          <path
            fillRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
            clipRule="evenodd"
          />
        </svg>
      </button>
      <div
        className={
          isNavExpanded ? "navigation-menu expanded" : "navigation-menu"
        }
      >
        <ul>
          <li>                
            <Link onClick={() => {setIsNavExpanded(false); }} to="/liveReload">Live</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/standingsReload">Standings</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/standingsPerTeamReload">Standings Per Team</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/gamesPerDateReload">Games Per Date</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/gamesPerTeamReload">Games Per Team And Season</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/searchPlayerReload">Search Player</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/searchTeamReload">Search Team</Link>
          </li>
          <li>
            <Link onClick={() => {setIsNavExpanded(false); }} to="/mostWinningFormationReload">Most Winning Formation</Link>
          </li>
        </ul>
      </div>
    </nav>
    </div>
    <div>
      <Routes>
          <Route path="/" element={ <Navigate replace to="/live"/> } />  
          <Route path="/liveReload" element={ <Navigate replace to="/live"/> } />
          <Route path="/standingsReload" element={ <Navigate replace to="/standings"/> } />
          <Route path="/standingsPerTeamReload" element={ <Navigate replace to="/standingsPerTeam"/> } />
          <Route path="/gamesPerDateReload" element={ <Navigate replace to="/gamesPerDate"/> } />
          <Route path="/gamesPerTeamReload" element={ <Navigate replace to="/gamesPerTeam"/> } />  
          <Route path="/searchPlayerReload" element={ <Navigate replace to="/searchPlayer"/> } />
          <Route path="/searchTeamReload" element={ <Navigate replace to="/searchTeam"/> } />
          <Route path="/mostWinningFormationReload" element={ <Navigate replace to="/mostWinningFormation"/> } />
          <Route path="/live" element={ <Live teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />        
          <Route path="/standings" element={ <StandingView teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />         
          <Route path="/standingsPerTeam" element={ <StandingsPerTeamView teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />        
          <Route path="/gamesPerDate" element={ <GamesPerDateView teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />
          <Route path="/gamesPerTeam" element={ <GamesPerTeam teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />  
          <Route path="/searchPlayer" element={ <SearchPlayer teamClicked={teamClicked} setTeamClicked={setTeamClicked} teamName = {teamName} setTeamName={setTeamName} /> } />    
          <Route path="/searchTeam" element={ <SearchTeam /> } />
          <Route path="/mostWinningFormation" element={ <WinningTeamView/> } />  
      </Routes>
    </div>
    </Router>
    </div>
    </ThemeProvider>
  );
}

export default App;