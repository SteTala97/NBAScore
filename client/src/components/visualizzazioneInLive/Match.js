import React from 'react'
import './Match.css'
import Team from './Team.js';
import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import LinearProgress from '@mui/material/LinearProgress';
import Box from '@mui/material/Box';
import Info from '@mui/icons-material/Info';
import Button from '@mui/material/Button';

function Progress(props) {
    const type = props.typeGame;
    if(type === "In Play"){
        return <LinearProgress color="success" />
    }else{
        return ""
    }
}

export default class Match extends React.Component {

    constructor(props) {

        super(props);

        this.state = {
            id: null, //identificare la partita e da passare tra componenti ed altri
            typeGame: "", //Not Started, live, Finished
            currentPeriods: null,
            totalPeriods: null,
            linescoreHome: null, //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
            linescoreVisitors: null, //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
            teamScoreHome: null,
            teamScoreVisitors: null,
            teamNameHome: null,
            teamLogoHome: null,
            teamCodeHome: null,
            teamNameVisitors: null,
            teamLogoVisitors: null,
            teamCodeVisitors: null,
            time: null
        }
    }

    render() {

        return (
            <TableRow
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
                <TableCell id="content-table-type" component="th" scope="row" align="center" sx={{width:'10%'}}>
                    {this.props.data === null ? this.state.typeGame : this.props.data.typeGame}
                    <Box sx={{ width: '10vh' }}>
                    <Progress typeGame = {this.props.data===null ? this.state.typeGame : this.props.data.typeGame}></Progress>
                    </Box>
                </TableCell>
                <TableCell id="content-table-logo" align="right" sx={{width:'15%'}}>
                    <img src={this.props.data===null ? this.state.teamLogoHome : this.props.data.teamLogoHome} className="team-logo selectable-team" alt="Logo Team" onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameHome) }}/>
                    <br></br><br></br>
                    <img src={this.props.data===null ? this.state.teamLogoVisitors : this.props.data.teamLogoVisitors} className="team-logo selectable-team" alt="Logo Team" onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameVisitors) }}/>
                </TableCell>
                <TableCell id="content-table-name" align="left" sx={{width:'15%'}}>
                    <Team name={this.props.data===null ? this.state.teamNameHome : this.props.data.teamNameHome} setTeamName={this.props.setTeamName} setTeamClicked={this.props.setTeamClicked} /> 
                    <br></br><br></br><br></br>
                    <Team name={this.props.data===null ? this.state.teamNameVisitors : this.props.data.teamNameVisitors} setTeamName={this.props.setTeamName} setTeamClicked={this.props.setTeamClicked} />
                </TableCell>
                <TableCell id="content-table-score" align="center" sx={{width:'10%'}}> 
                    {this.props.data===null || this.props.data.teamScoreHome===null ? " - " : this.props.data.teamScoreHome}
                    <br></br><br></br><br></br>
                    {this.props.data===null || this.props.data.teamScoreVisitors===null ? " - " : this.props.data.teamScoreVisitors}
                </TableCell>
                <TableCell id="content-table-time" align="center" sx={{width:'10%'}}>{this.props.data === null ? this.state.time : this.props.data.time}</TableCell>
                <TableCell id="content-table-quarter" align="center" sx={{width:'10%'}}>{this.props.data===null ? this.state.currentPeriods : this.props.data.currentPeriods} of {this.props.data===null ? this.state.totalPeriods : this.props.data.totalPeriods}</TableCell>
                <TableCell id="content-table-pointsPerQuarter" align="center" sx={{width:'15%'}}> 
                    [
                    {this.props.data===null || this.props.data.typeGame==="Scheduled" || this.props.data.linescoreHome.length===0 ? "\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0" : this.props.data.linescoreHome.map(
                        (scorePeriod) => " "+scorePeriod+" ")}
                    ]
                    <br></br><br></br><br></br>
                    [
                    {this.props.data===null || this.props.data.typeGame==="Scheduled" || this.props.data.linescoreVisitors.length===0 ? "\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0" : this.props.data.linescoreVisitors.map(
                        (scorePeriod) => " "+scorePeriod+" ")}
                    ]
                </TableCell>
                <TableCell id="content-table-button" align="center" sx={{width:'10%'}}> 
                    <div>
                    <Button onClick={() => {this.props.setButtonClicked(true); this.props.setGame_Id(this.props.data.id) }} variant="contained" color="success" endIcon={<Info fontSize='medium'/>} className="more-info-button"> 
                        More Info
                    </Button>
                    </div>
                </TableCell>
            </TableRow>
        );
    
    }
    
}

