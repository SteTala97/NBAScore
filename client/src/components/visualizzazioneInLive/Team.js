import React from 'react'
import './Team.css'


export default class Team extends React.Component {

    render() {
        return (
        <div id="content-team-name" className='team-text'>
            <div className="selectable-team" onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.name) }}>
                {this.props.name}
            </div>
        </div>
        );

    }
}