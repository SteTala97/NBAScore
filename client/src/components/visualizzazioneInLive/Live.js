import React, { useState, useEffect } from 'react';
import './Live.css';
import Match from './Match.js';
import GameInfo from '../visualizzazioneInformazioniPartita/GameInfo';
import TeamInfo from '../visualizzazioneInformazioniSquadra/TeamInfo';

import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import MuiAlert from '@mui/material/Alert';


const axios = require('axios');

const Alert = React.forwardRef(function Alert(props, ref) {
  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
});

function CheckEmpty(props) {
  const empty = props.totalGames.length===0 ? true : false;
  if(props.callDone){
    if(empty){
      return <Alert id="alert-no-match" severity="warning">There are no games at the moment!</Alert>
    }else{
      return <TableGames id="table-games" totalGames = {props.totalGames} setTeamClicked = {props.setTeamClicked} teamClicked={props.teamClicked} setTeamName={props.setTeamName} teamName={props.teamName}></TableGames>
    }
  }else{
    return <React.Fragment></React.Fragment>
  }
}

function TableGames(props) {

  const [buttonClicked, setButtonClicked] = useState(false);
  const [game_Id, setGame_Id] = useState("");

  
  useEffect(() => {
    props.setTeamClicked(false);
    props.setTeamName("");
  }, []);

  return(
    <React.Fragment>
    {buttonClicked === true ?
      <GameInfo 
          id = {game_Id}
          teamName = {props.teamName}
          teamClicked = {props.teamClicked}
          setTeamClicked={props.setTeamClicked} 
          setTeamName={props.setTeamName}
      /> 
      :
      props.teamClicked === true ?
        <TeamInfo teamName = {props.teamName} />
        :
        <TableContainer className="LiveTableContainer">
          <Table id="table" sx={{ minWidth: 650, maxWidth: 1000}} aria-label="simple table" className="LiveTable">
            <TableHead>
              <TableRow key="1">
                <TableCell id="title-table-type" align="center"><strong>Status</strong></TableCell>
                <TableCell id="title-table-teams" align="center" colSpan={2}><strong>Teams</strong></TableCell>
                <TableCell id="title-table-points" align="center"><strong>Points</strong></TableCell>
                <TableCell id="title-table-time" align="center"><strong>Time</strong></TableCell>
                <TableCell id="title-table-time" align="center"><strong>Quarter</strong></TableCell>
                <TableCell id="title-table-linescore" align="center"><strong>Points per Quarter</strong></TableCell>
                <TableCell id="title-table-button" align="center">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
                {props.totalGames.map((game) => (
                  <Match key={game.id.toString()} data={game} setButtonClicked={setButtonClicked} setGame_Id={setGame_Id} setTeamClicked={props.setTeamClicked} setTeamName={props.setTeamName} teamClicked={props.teamClicked} teamName={props.teamName}></Match>
                ))}
                <TableRow>
                </TableRow>
            </TableBody>
          </Table>
        </TableContainer>
      }
      </React.Fragment>
  );
}

export default class Live extends React.Component {

  componentDidMount(props) {
    this.interval = setInterval(() => this.tick(), 1000);
    this.fetch();
  }

  componentWillUnmount() {
    clearInterval(this.interval);
    this.setState(state => ({
      callDone: false
    }));
  }

  tick() {
    this.setState(state => ({
      seconds: state.seconds + 1
    }));
    if(this.state.seconds%120 === 0){
      this.fetch()
    }
  }

  fetch(){
    var context = this;

    let ts = Date.now();

    let date_ob = new Date(ts);
    let date = date_ob.getDate();
    let month = date_ob.getMonth() + 1;
    let year = date_ob.getFullYear();
    if(date < 10)
      date = '0'+date;
    
    if(month < 10)
      month = '0'+month;
    let str = year + '-' + month + '-' + date;
    axios.get("https://nbascore.herokuapp.com/gamesToday", { params: { data: str} })
      .then(function (response) {
        context.setState({
          games: response.data,
          callDone: true
        });
      })
      .catch(function (error) {
      })
  }

  constructor(props) {

    super(props);

    this.state = {
      games: [],
      seconds: parseInt(props.startTimeInSeconds, 10) || 0,
      callDone: false
    }
  }

  render() {
    return (
        <CheckEmpty totalGames = {this.state.games} callDone = {this.state.callDone} teamClicked={this.props.teamClicked} setTeamClicked = {this.props.setTeamClicked} teamName={this.props.teamName} setTeamName={this.props.setTeamName}></CheckEmpty>        
    );
  }

}