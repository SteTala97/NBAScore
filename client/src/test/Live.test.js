import React from 'react';

//import { expect } from 'chai';
import { shallowToJson } from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import App from '../App';
import Live from '../components/visualizzazioneInLive/Live';
import Match from '../components/visualizzazioneInLive/Match';
import Team from '../components/visualizzazioneInLive/Team';

describe("Test rendering componenti live", () => {
  
    const testMatch = 
      { 
        id: 1, //identificare la partita e da passare tra componenti ed altri
        typeGame: 'Finished', //Not Started, live, Finished
        currentPeriods: 4,
        totalPeriods: 4,
        linescoreHome: ["36", "25", "36", "24"], //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
        linescoreVisitors: ["36", "25", "36", "24"], //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
        teamScoreHome: 23,
        teamScoreVisitors: 40,
        teamNameHome: 'Team 1',
        teamLogoHome: '',
        teamCodeHome: 'ESP',
        teamNameVisitors: 'Team 2',
        teamLogoVisitors: '',
        teamCodeVisitors: 'SDF'
      };

    const testTeam = "Team 1";

    let otherProps = {
      teamClicked: false, 
      teamName: ""
    };

    it("Rendering del componente App senza crashare", () => {
        shallow(<App/>);
    });

    it("Rendering del componente Live senza crashare", () => {
        const wrapper = mount(<Live teamClicked={otherProps.teamClicked} teamName={otherProps.teamName} />);
    });

    it("Rendering del componente Match senza crashare", () => {
        const wrapper = mount(<Match key={testMatch.id.toString()} data={testMatch} teamClicked={otherProps.teamClicked} teamName={otherProps.teamName} />);
    });

    it("Rendering del componente Match senza crashare", () => {
      const wrapper = mount(<Team name={testTeam}/>);
    });

});

// Test that the rendering does not change when the same input data is given
describe("Test degli Snapshot", () => {

  const testMatch = 
      { 
        id: 1, //identificare la partita e da passare tra componenti ed altri
        typeGame: 'Finished', //Not Started, live, Finished
        currentPeriods: 4,
        totalPeriods: 4,
        linescoreHome: ["36", "25", "36", "24"], //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
        linescoreVisitors: ["36", "25", "36", "24"], //questo è un array composto [0:"36", 1:"25", 2:"36", 3: "24"]
        teamScoreHome: 23,
        teamScoreVisitors: 40,
        teamNameHome: 'Team 1',
        teamLogoHome: '',
        teamCodeHome: 'ESP',
        teamNameVisitors: 'Team 2',
        teamLogoVisitors: '',
        teamCodeVisitors: 'SDF'
      };

  const testTeam = "Team 1";

  let otherProps = {
    teamClicked: false, 
    teamName: ""
  };

  it("Rendering del componente Live correttamente", () => {
      const tree = shallow(<Live teamClicked={otherProps.teamClicked} teamName={otherProps.teamName}/>);
      expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("Rendering del componente Live con TeamInfo", () => {
    otherProps.teamClicked = true;
    otherProps.teamName = "Miami Heat";
    const tree = shallow(<Live teamClicked={otherProps.teamClicked} teamName={otherProps.teamName}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("Rendering del comp. Match senza passare dati", () => {
      const tree = shallow(<Match data={null}/>);
      expect(shallowToJson(tree)).toMatchSnapshot();
  });
  it("Rendering del comp. Match correttamente", () => {
      const tree = shallow(<Match key={testMatch.id.toString()} data={testMatch} teamClicked={otherProps.teamClicked} teamName={otherProps.teamName}/>);
      expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("Rendering del comp. Match con TeamInfo", () => {
    otherProps.teamClicked = true;
    otherProps.teamName = "Miami Heat";
    const tree = shallow(<Match key={testMatch.id.toString()} data={testMatch} teamClicked={otherProps.teamClicked} teamName={otherProps.teamName}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("Rendering del comp. Team senza passare valori", () => {
      const tree = shallow(<Team />);
      expect(shallowToJson(tree)).toMatchSnapshot();
  });
  it("Rendering del componente Team correttamente", () => {
      const tree = shallow(<Team name={testTeam}/>);
      expect(shallowToJson(tree)).toMatchSnapshot();
  });
})
