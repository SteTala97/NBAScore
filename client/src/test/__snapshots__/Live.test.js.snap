// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test degli Snapshot Rendering del comp. Match con TeamInfo 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    Finished
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame="Finished"
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src=""
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src=""
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <Team
      name="Team 1"
    />
    <br />
    <br />
    <br />
    <Team
      name="Team 2"
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    23
    <br />
    <br />
    <br />
    40
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-time"
    sx={
      Object {
        "width": "10%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    4
     of 
    4
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
     36 
     25 
     36 
     24 
    ]
    <br />
    <br />
    <br />
    [
     36 
     25 
     36 
     24 
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Test degli Snapshot Rendering del comp. Match correttamente 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    Finished
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame="Finished"
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src=""
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src=""
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <Team
      name="Team 1"
    />
    <br />
    <br />
    <br />
    <Team
      name="Team 2"
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    23
    <br />
    <br />
    <br />
    40
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-time"
    sx={
      Object {
        "width": "10%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    4
     of 
    4
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
     36 
     25 
     36 
     24 
    ]
    <br />
    <br />
    <br />
    [
     36 
     25 
     36 
     24 
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Test degli Snapshot Rendering del comp. Match senza passare dati 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame=""
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src={null}
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src={null}
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <Team
      name={null}
    />
    <br />
    <br />
    <br />
    <Team
      name={null}
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
     - 
    <br />
    <br />
    <br />
     - 
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-time"
    sx={
      Object {
        "width": "10%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
     of 
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
       -   -   -   -   
    ]
    <br />
    <br />
    <br />
    [
       -   -   -   -   
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Test degli Snapshot Rendering del comp. Team senza passare valori 1`] = `
<div
  className="team-text"
  id="content-team-name"
>
  <div
    className="selectable-team"
    onClick={[Function]}
  />
</div>
`;

exports[`Test degli Snapshot Rendering del componente Live con TeamInfo 1`] = `
<CheckEmpty
  callDone={false}
  teamClicked={true}
  teamName="Miami Heat"
  totalGames={Array []}
/>
`;

exports[`Test degli Snapshot Rendering del componente Live correttamente 1`] = `
<CheckEmpty
  callDone={false}
  teamClicked={false}
  teamName=""
  totalGames={Array []}
/>
`;

exports[`Test degli Snapshot Rendering del componente Team correttamente 1`] = `
<div
  className="team-text"
  id="content-team-name"
>
  <div
    className="selectable-team"
    onClick={[Function]}
  >
    Team 1
  </div>
</div>
`;
